rules/mapping.go

```diff
@@ -302,20 +329,17 @@ func (mc *mappingRule) activeIndex(timeNanos int64) int {
 	return idx
 }
 
-func (mc *mappingRule) history() []MappingRuleView {
-	views := make([]MappingRuleView, len(mc.snapshots))
-	for i, s := range mc.snapshots {
-		views[i] = newMappingRuleView(mc.uuid, *s)
+func (mc *mappingRule) history() []*MappingRuleView {
+	views := make([]*MappingRuleView, len(mc.snapshots))
+	for i := 0; i < len(mc.snapshots); i++ {
+		mrs, _ := newMappingRuleView(mc, i)
```

Comment:
nit: change this to mc.mappingRuleView(idx int) *MappingRuleView?

Reply:
also probably should handle the error here

Reply:
Sure